pkg/healthcheck/checks.go

```diff
@@ -8,16 +8,11 @@ import (
 )
 
 // CheckHTTPStatus checks if the given HTTP request responds with the expected status code
-func CheckHTTPStatus(ctx context.Context, method string, url string, wantStatusCode int, timeout time.Duration) error {
-	httpClient := http.Client{
-		Timeout: timeout,
-	}
-
+func CheckHTTPStatus(ctx context.Context, httpClient *http.Client, method string, url string, wantStatusCode int, timeout time.Duration) error {
```

Comment:
We should not accept a pointer type here but an interface that can be mocked appropriately